Add SLE binaries

jimmykarily opened the pull request:
Signed-off-by: André Duffeck <aduffeck@suse.com>

Commit:
Add SLE binaries

Signed-off-by: André Duffeck <aduffeck@suse.com>

alexjh commented:
There is something wrong with this buildpack / rootfs. 

Using https://github.com/cloudfoundry-samples/fib-cpu as an example app. The Gemfile needs to be changed to get a supported version, but otherwise it should work. It works on the `opensuse42` stack.

1. Looks like we need the `logger` utility?
2. There is some confusion about the Ruby version? `Your Ruby version is 2.3.3, but your Gemfile specified 2.2.7` Perhaps a `PATH` problem where it's picking up the system ruby?

`cf push -s sle12`

results in:

```
Creating container
Successfully created container
Downloading app package...
Downloaded app package (534.9K)
Staging...
-------> Buildpack version 1.6.40
       Downloaded [file:///tmp/buildpacks/de34a5106f27cfb1b4b8b0c52299c59e/dependencies/https___cf-sle12.s3.amazonaws.com_dependencies_bundler_bundler-1.15.0-788e45a5.tgz]
-----> Supplying Ruby/Rack
sh: logger: command not found
       Downloaded [file:///tmp/buildpacks/de34a5106f27cfb1b4b8b0c52299c59e/dependencies/https___cf-sle12.s3.amazonaws.com_dependencies_ruby_ruby-2.2.7-linux-x64-67bc4aae.tgz]
-----> Using Ruby version: ruby-2.2.7
sh: logger: command not found
-------> Buildpack version 1.6.40
-----> Finalizing Ruby/Rack
sh: logger: command not found
sh: logger: command not found
-----> Installing dependencies using bundler 1.15.0
       Running: bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin --jobs=4 --retry=4 --deployment
       Your Ruby version is 2.3.3, but your Gemfile specified 2.2.7
sh: logger: command not found
       Bundler Output: Your Ruby version is 2.3.3, but your Gemfile specified 2.2.7
sh: logger: command not found
sh: logger: command not found
 !
 !     Failed to install gems via Bundler.
 !
Failed to compile droplet
Exit status 223
Staging failed: Exited with status 223
Destroying container
Successfully destroyed container

FAILED
Error restarting application: BuildpackCompileFailed

TIP: use 'cf logs fib-cpu --recent' for more information
```

When the Ruby version is set to 2.3.3, the ruby headers aren't found:

```
-----> Installing dependencies using bundler 1.15.0
       Running: bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin --jobs=4 --retry=4 --deployment
       Installing rack 1.6.4
       Installing kgio 2.10.0 with native extensions
       Installing raindrops 0.15.0 with native extensions
       Using bundler 1.15.0
       Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
       current directory:
       /tmp/app/vendor/bundle/ruby/2.3.0/gems/raindrops-0.15.0/ext/raindrops
       /usr/bin/ruby.ruby2.3 -r ./siteconf20171017-116-gonn96.rb extconf.rb
       mkmf.rb can't find header files for ruby at /usr/lib64/ruby/include/ruby.h
       extconf failed, exit code 1
       Gem files will remain installed in
       /tmp/app/vendor/bundle/ruby/2.3.0/gems/raindrops-0.15.0 for inspection.
       Results logged to
       /tmp/app/vendor/bundle/ruby/2.3.0/extensions/x86_64-linux/2.3.0/raindrops-0.15.0/gem_make.out
       An error occurred while installing raindrops (0.15.0), and Bundler cannot
       continue.
       Make sure that `gem install raindrops -v '0.15.0'` succeeds before bundling.
       In Gemfile:
       unicorn was resolved to 4.9.0, which depends on
       raindrops
```

alexjh commented:
Looks like our problem is that shebang line is pointing to the versioned Ruby: `#!/usr/bin/env ruby.ruby2.3`

The Ruby binary assets will have to be regenerated after we figure out how to fix this.

jimmykarily commented:
Fixed the object_id and the signature and squashed the commits. Also deployed [a ruby application](https://github.com/Stackato-Apps/sinatra-env) with this buildpack and worked fine. Only had to use a supported version (tried with 2.4.1 and 2.2.7) and update the eventmachine gem to a newer version because [of a bug](https://github.com/Stackato-Apps/sinatra-env).
